xonsh/main.py

```diff
@@ -19,6 +19,39 @@
 from xonsh.xonfig import xonfig_main
 from xonsh.lazyimps import pygments, pyghooks
 from xonsh.imphooks import install_hook
+from xonsh.events import events
+
+
+events.transmogrify('on_post_init', 'LoadEvent')
+events.doc('on_post_init', """
+on_post_init() -> None
+
+Fired after all initialization is finished and we're ready to do work.
+
+NOTE: This is fired before the wizard is automatically started.
+""")
+
+events.transmogrify('on_pre_cmdloop', 'LoadEvent')
+events.doc('on_pre_cmdloop', """
+on_pre_cmdloop() -> None
```

Comment:
I think it might be a little different (depending on when things are registered) because all of the `__xonsh_*__` builtins may be unloaded by the time atexit is called.  If there was a post cmdloop we would guarantee that the state was still sufficiently xonshy.